scheduler/data-flow/src/main/kotlin/io/seldon/dataflow/kafka/Configuration.kt

```diff
@@ -75,21 +75,25 @@ private fun getSecurityProperties(params: KafkaStreamsParams): Properties {
 }
 
 private fun getSslProperties(params: KafkaStreamsParams): Properties {
-    return Properties().apply {
-        with(params.security.certConfig) {
-            if (clientSecret != "" && brokerSecret != "") {
-                K8sCertSecretsProvider.downloadCertsFromSecrets(this)
-            }
-        }
+    val certConfig = params.security.certConfig
 
-        val (keyStoreConfig, trustStoreConfig) = Provider.keyStoresFromCertificates(params.security.certConfig)
-        this[SslConfigs.SSL_KEYSTORE_LOCATION_CONFIG] = keyStoreConfig.keyStoreLocation
-        this[SslConfigs.SSL_KEYSTORE_PASSWORD_CONFIG] = keyStoreConfig.keyStorePassword
-        this[SslConfigs.SSL_KEY_PASSWORD_CONFIG] = keyStoreConfig.keyStorePassword
+    if (certConfig.brokerSecret != "" || certConfig.clientSecret != "") {
+        K8sCertSecretsProvider.downloadCertsFromSecrets(certConfig)
+    }
+
+    return Properties().apply {
+        val trustStoreConfig = Provider.trustStoreFromCertificates(certConfig)
         this[SslConfigs.SSL_TRUSTSTORE_LOCATION_CONFIG] = trustStoreConfig.trustStoreLocation
         this[SslConfigs.SSL_TRUSTSTORE_PASSWORD_CONFIG] = trustStoreConfig.trustStorePassword
-        this[SslConfigs.SSL_ENDPOINT_IDENTIFICATION_ALGORITHM_CONFIG] =
-            params.security.certConfig.endpointIdentificationAlgorithm
+
+        if (certConfig.clientSecret.isNotEmpty()) {
+            val keyStoreConfig = Provider.keyStoreFromCertificates(certConfig)
+            this[SslConfigs.SSL_KEYSTORE_LOCATION_CONFIG] = keyStoreConfig.keyStoreLocation
+            this[SslConfigs.SSL_KEYSTORE_PASSWORD_CONFIG] = keyStoreConfig.keyStorePassword
+            this[SslConfigs.SSL_KEY_PASSWORD_CONFIG] = keyStoreConfig.keyStorePassword
+        }
+
+        this[SslConfigs.SSL_ENDPOINT_IDENTIFICATION_ALGORITHM_CONFIG] = certConfig.endpointIdentificationAlgorithm
```

Comment:
Does the `endpointIdentificationAlgorithm` key in Kotlin follow similar rules to the config in Go? As in,

- If not present, default to `ssl`
- If present but empty, default to `plaintext`
- Otherwise, use the value

Reply:
From the [documentation](https://docs.confluent.io/platform/current/kafka/authentication_ssl.html#id1)  

> The endpoint identification algorithm used by clients to validate server host name. The default value is https. Clients including client connections created by the broker for interbroker communication verify that the broker host name matches the host name in the broker’s certificate. Disable server host name verification by setting ssl.endpoint.identification.algorithm to an empty string.

The thing was that Golang client errors when you set it to an empty string, hence I believe where the value `none` comes from. 

The behaviour should therefore be consistent - worth to note that we always set the environmental variable for in our helm deployment.